### PR TITLE
[SDPA-4924] Fix Link text doesn't wrap in accordions

### PR DESCRIPTION
**Pull request by mayurngondhkar**
## Motivation and Context

**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-4924

## Changed

1. Link text doesn't wrap

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist

- [ ] I've added relevant changes to the documentation.
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] My change requires a template update for create-ripple-app.
- [ ] I have added template update script for next release.

### CODE REVIEW

**Comment by mayurngondhkar** [comment]
Hi @rubyllamadora,

Can you please review this PR.

https://app.pr-964.ripple.sdp2.sdp.vic.gov.au/
https://app.pr-964.ripple.sdp2.sdp.vic.gov.au/clone-demo-landing-page

Image:
image

image


https://digital-engagement.atlassian.net/browse/SDPA-4924



Thanks

File: packages/components/Atoms/Icon/TextIcon.vue [hunk 1]
```diff
@@ -5,15 +5,22 @@
...
   </span>
   <span v-else-if="text && symbol && placement === 'after'">
-    <span v-if="textWordCount > 1">{{ textWithoutLastWord }}</span>
-    <span v-if="textWordCount > 1" class="rpl-text-icon__group"> {{ textLastWord }}<rpl-icon v-bind="iconProps" /></span>
+    <span v-if="textWordCount > 1">
+      <span>{{ textWithoutLastWord }}</span>
+      <span class="rpl-text-icon__group"> {{ textLastWord }}<rpl-icon v-bind="iconProps" /></span>
+    </span>
+    <span v-else-if="textWordCount === 1 && textIsAbsoluteUrl" class="rpl-text-icon__url-icon">
+      <div class="rpl-text-icon__absolute-url">{{ text }}</div>
```
Comment (rubyllamadora) [inline packages/components/Atoms/Icon/TextIcon.vue:13]: Hmm this is tricky as text props can accept a group of words or just URL or both. Can you try this css setting 
```
overflow-wrap: break-word;
word-wrap: break-word;
```
It should break long words and respect the parent's width.
I found sample code in here https://codepen.io/team/css-tricks/pen/RWaNxr

Comment (rubyllamadora) [reply packages/components/Atoms/Icon/TextIcon.vue:13]: Also can we update unit test with this test case please `text-icon.test.js`

File: packages/components/Atoms/Icon/TextIcon.vue [hunk 2]
```diff
@@ -86,5 +102,21 @@ export default {
     &--after {
       margin: $rpl-text-link-after-margin;
     }
+
+    &__url-icon {
+      display: flex;
+    }
+
+    &__absolute-url {
+      display: block;
```
Comment (rubyllamadora) [inline packages/components/Atoms/Icon/TextIcon.vue:111]: This will bring down the whole text to the next line. Try inline-block?

**Review by rubyllamadora — CHANGES_REQUESTED** [review]
Hey @mayurngondhkar, good effort on trying to fix the issue however I'm still not convinced to add custom absolute_url validation to the component. I feel like the component should stand to its Text+Icon. And the main issue here is when the TextIcon is wrapped in an Accordion/Markup - the URL just doesn't break or respect the max-width set in markup__inner.

Can you try removing 
```
&__group {
  white-space: nowrap;
}
```
in https://github.com/dpc-sdp/ripple/pull/964/files#diff-5744049b83242975493c5079979620b8d1994802977a578977a9e9a1956c2aefR95

Hopefully it works so there's no need to add custom validation (:

File: packages/components/Atoms/Icon/TextIcon.vue (hunk 2 above)
Comment (dylankelly) [reply packages/components/Atoms/Icon/TextIcon.vue:111]: @rubyllamadora @mayurngondhkar Not required, but I reckon this is a good candidate for a mixin - See Chris Coyiers https://css-tricks.com/snippets/css/prevent-long-urls-from-breaking-out-of-container/#for-the-scss-inclined

`@mixin rpl_word_wrap`

File: packages/components/Atoms/Icon/TextIcon.vue [hunk 3]
```diff
@@ -63,6 +76,9 @@ export default {
         size: this.size,
         class: `rpl-text-icon--${this.placement}`
       }
+    },
+    textIsAbsoluteUrl: function () {
```
Comment (dylankelly) [inline packages/components/Atoms/Icon/TextIcon.vue:80]: if `isAbsoluteUrl` is a function, no need to wrap in another function. 
```
<template>
<span v-if="isAbsoluteUrl(text)">something</span> 
</template>
<script>
methods: {
  isAbsoluteUrl,
}
</script>
```

**Review by dylankelly — CHANGES_REQUESTED** [review]
> Hey @mayurngondhkar, good effort on trying to fix the issue however I'm still not convinced to add custom absolute_url validation to the component. I feel like the component should stand to its Text+Icon. And the main issue here is when the TextIcon is wrapped in an Accordion/Markup - the URL just doesn't break or respect the max-width set in markup__inner.
> 
> Can you try removing
> 
> ```
> &__group {
>   white-space: nowrap;
> }
> ```
> 
> in https://github.com/dpc-sdp/ripple/blob/develop/packages/components/Atoms/Icon/TextIcon.vue#L78
> 
> Hopefully it works so there's no need to add custom validation (:

I agree with @rubyllamadora - not convinced this is needed.

File: packages/components/Atoms/Icon/TextIcon.vue (hunk 3 above)
Comment (mayurngondhkar) [reply packages/components/Atoms/Icon/TextIcon.vue:80]: I spoke to Jess and she was ok with the text wrapping to the next line.

Thanks @dylankelly and @rubyllamadora

**Comment by mayurngondhkar** [comment]
@rubyllamadora please have a look at the PR when you get a chance. Thanks

File: packages/components/Atoms/Icon/TextIcon.vue (hunk 3 above)
Comment (dylankelly) [reply packages/components/Atoms/Icon/TextIcon.vue]: @mayurngondhkar have another look at my comment - im talking about how you are adding this function into the vue component

Comment (dylankelly) [reply packages/components/Atoms/Icon/TextIcon.vue]: ok i see you have removed this 👍

**Review by rubyllamadora — APPROVED** [review]

**Pull request merged**